pkg/client/connector/traffic_manager.go

```diff
@@ -499,3 +511,102 @@ func (tm *trafficManager) uninstall(c context.Context, ur *rpc.UninstallRequest)
 	}
 	return result, nil
 }
+
+var svcCIDRrx = regexp.MustCompile(`range of valid IPs is (.*)$`)
+
+// getClusterCIDRs finds the service CIDR and the pod CIDRs of all nodes in the cluster
+func (tm *trafficManager) getOutboundInfo(c context.Context, mgrPort int32) (*daemon.OutboundInfo, error) {
+	// Get all nodes
+	var nodes []kates.Node
+	var cidr *net.IPNet
+
+	dnsServices := []metav1.ObjectMeta{
+		{
+			Name:      "kube-dns",
+			Namespace: "kube-system",
+		},
+		{
+			Name:      "dns-default",
+			Namespace: "openshift-dns",
+		},
+	}
+	var kubeDNS net.IP
+	for _, dnsService := range dnsServices {
+		svc := kates.Service{TypeMeta: metav1.TypeMeta{Kind: "Service"}, ObjectMeta: dnsService}
+		if err := tm.client.Get(c, &svc, &svc); err == nil {
+			kubeDNS = iputil.Parse(svc.Spec.ClusterIP)
+			break
+		}
+	}
+
+	if kubeDNS == nil {
+		return nil, fmt.Errorf("failed to get IP of cluster's DNS service")
+	}
+
+	// make an attempt to create a service with ClusterIP that is out of range and then
+	// check the error message for the correct range as suggested tin the second answer here:
+	//   https://stackoverflow.com/questions/44190607/how-do-you-find-the-cluster-service-cidr-of-a-kubernetes-cluster
+	// This requires an additional permission to create a service, which we might not have
+	svc := kates.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "t2-tst-dummy",
+		},
+		Spec: v1.ServiceSpec{
+			Ports:     []kates.ServicePort{{Port: 443}},
+			ClusterIP: "1.1.1.1",
+		},
+	}
+
+	var serviceSubnet *daemon.IPNet
+	if err := tm.client.Create(c, &svc, &svc); err != nil {
+		if match := svcCIDRrx.FindStringSubmatch(err.Error()); match != nil {
+			_, cidr, err = net.ParseCIDR(match[1])
+			if err != nil {
+				dlog.Errorf(c, "unable to parse service CIDR %q", match[1])
+			} else {
+				serviceSubnet = iputil.IPNetToRPC(cidr)
+			}
+		} else {
+			dlog.Debugf(c, "Probably insufficient permissions to attempt dummy service creation in order to find service subnet: %v", err)
+		}
+	}
+
+	if serviceSubnet == nil {
+		// Using a "kubectl cluster-info dump" or scanning all services generates a lot of unwanted traffic
+		// and would quite possibly also require elevated permissions, so instead, we derive the service subnet
+		// from the kubeDNS IP. This is cheating but a cluster may only have one service subnet and the mask is
+		// unlikely to cover less than half the bits.
+		dlog.Debugf(c, "Deriving serviceSubnet from %s (the IP of kube-dns.kube-system)", kubeDNS)
+		bits := len(kubeDNS) * 8
+		ones := bits / 2
+		mask := net.CIDRMask(ones, bits) // will yield a 16 bit mask on IPv4 and 64 bit mask on IPv6.
+		serviceSubnet = &daemon.IPNet{Ip: kubeDNS.Mask(mask), Mask: int32(ones)}
+	}
+
+	if err := tm.client.List(c, kates.Query{Kind: "Node"}, &nodes); err != nil {
```

Comment:
Edit: my 2nd suggestion might not even apply since we have an open issue to provide manager yaml that people can use to deploy the traffic-manager, so perhaps there isn't a way around it.. I'll sleep on it.

----

I haven't thought about this much, but is there an alternative way for us to get this information without the nodes.. _or_ could we calculate this when the operator is launching the traffic-manager  and then get that value via gRPC call to the manger?  

I only say this bc requiring users to list nodes is a cluster permission and, at least at my last job, that wasn't given to developers and was rarely even given to my team as DevOps.  So it's kind a substantial permission escalation (imo). But if this is the only way we can get this info, then I think it's a worthy trade-off and maybe we come up w/ some workaround or other way to handle people who _can't_ list nodes.

Reply:
I would like to get this info from the traffic-manager (I even have a branch where I'm testing it), but that in turn requires an RBAC setup for the manager. A default setup won't let me use kates at all. While I think that's something that we really will want, I though it was beyond the scope of the work I was doing. But perhaps that's a necessary step.

Reply:
I re-requested your review because you are requesting changes. If it's related to this then I suggest we fix this when we move the subnet discovery over to the traffic-manager service.